dora/src/masm/x64.rs

```diff
@@ -417,6 +418,19 @@ impl MacroAssembler {
         }
     }
 
+    pub fn int_shlx(&mut self, mode: MachineMode, dest: Reg, lhs: Reg, rhs: Reg) {
+        let x64 = match mode {
+            MachineMode::Int32 => 0,
+            MachineMode::Int64 => 1,
+            _ => unimplemented!(),
+        };
```

Comment:
(Not relevant for this commit.) Would it make sense in general to turn the `x64` snippet into a function instead of duplicating it in many places?

Reply:
Yes, I want to refactor the whole Assembler (see dora/src/asm.rs).